osquery/tables/system/windows/startup_items.cpp

```diff
@@ -57,6 +58,26 @@ const std::set<std::string> kStartupStatusRegKeys = {
 // Starts with 0[0-9] but not followed by all 0s
 const auto kStartupDisabledRegex = boost::regex("^0[0-9](?!0+$).*$");
 
+static inline void parseStartupPath(const std::string& path, Row& r) {
+  if (path.find('\"') == std::string::npos) {
+    r["path"] = path;
+  } else {
+    boost::tokenizer<boost::escaped_list_separator<TCHAR>> tokens(
+        path,
+        boost::escaped_list_separator<TCHAR>(
+            std::string(""), std::string(" "), std::string("\"\'")));
+    for (auto& tok = tokens.begin(); tok != tokens.end(); ++tok) {
```

Comment:
oh wait this was for a reason. The range-based loop returns a dereferenced pointer (i.e. tok is a std::string instead of a std::string *). This would be fine except I need to compare the pointer to the iterator's begin() pointer to determine whether I'm on the first token or not.

I could alternately do the first step outside of the loop, advance the iterator, and then use a range-based loop, but not sure that's any cleaner, unless there is an efficiency win?